coffee_lib/src/repository.rs

```diff
@@ -30,6 +30,9 @@ pub trait Repository: Any {
     /// recover the repository from the commit id.
     async fn recover(&mut self) -> Result<(), CoffeeError>;
 
+    /// switch to the specified branch.
+    async fn switch_branch(&mut self, branch_name: &str) -> Result<(), CoffeeError>;
```

Comment:
```suggestion
```

Does not maske sense have the `switch_branch` inside the repository interface, it can be just an util. For example look at the `repository.upgrade` code